ext/drreg/drreg.c

```diff
@@ -1719,37 +2613,74 @@ drreg_event_restore_state(void *drcontext, bool restore_memory,
         return true; /* fault not in cache */
     for (reg = DR_REG_START_GPR; reg <= DR_REG_STOP_GPR; reg++)
         spilled_to[GPR_IDX(reg)] = MAX_SPILLS;
+#ifdef X86
+    for (reg = DR_REG_APPLICABLE_START_SIMD; reg <= DR_REG_APPLICABLE_STOP_SIMD; reg++)
+        spilled_simd_to[SIMD_IDX(reg)] = MAX_SIMD_SPILLS;
+    for (slot = 0; slot < MAX_SIMD_SPILLS; ++slot)
+        simd_slot_use[slot] = DR_REG_NULL;
+#endif
+
     LOG(drcontext, DR_LOG_ALL, 3,
         "%s: processing fault @" PFX ": decoding from " PFX "\n", __FUNCTION__,
         info->raw_mcontext->pc, pc);
     instr_init(drcontext, &inst);
+    instr_init(drcontext, &next_inst);
     while (pc < info->raw_mcontext->pc) {
         instr_reset(drcontext, &inst);
+        instr_reset(drcontext, &next_inst);
         prev_pc = pc;
         pc = decode(drcontext, pc, &inst);
-
-        if (is_our_spill_or_restore(drcontext, &inst, &spill, &reg, &slot, &offs)) {
+        decode(drcontext, pc, &next_inst);
```

Comment:
Check for pc==NULL to avoid a crash here.  Good to check return value of this one too.

Reply:
I'm actually not sure about whether we should assert, or return true. What if somebody did sth fancy and placed something invalid into the code cache? I suggest we return true if decode() == NULL.

Reply:
I don't have any opinion on this. If we are indeed going to return true, we should at least add a log entry. I'll fix this issue once agreed.

Reply:
Any update on this please?

Reply:
I would argue that somebody could do sth fancy and e.g. instrument the code cache with sth undecodable. Hence we can't assert or abort. We also can't return false, because this incorrectly shortcuts other callbacks. So we need to check for pc == NULL and return true. Yes you could add a LOG entry.

Reply:
Done.